Implement macros for quote and image section

britneywwc opened the pull request:
## Done

- Add macros for quote and image section
- Implemented example use of macros on index page

## QA

- Go to https://ubuntu-com-14149.demos.haus/case-studies
- See that quotes and image section design matches with [Figma](https://www.figma.com/design/AeIJ3GsqnJCHtiBYhkTuTV/24.10-Case-study-template?node-id=1-716&t=ocgEedog4898THvX-0)

## Issue / Card

Fixes [WD-13122](https://warthogs.atlassian.net/browse/WD-13122) and [WD-13123](https://warthogs.atlassian.net/browse/WD-13123)

## Screenshots

[If relevant, please include a screenshot.]


## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)


[WD-13122]: https://warthogs.atlassian.net/browse/WD-13122?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WD-13123]: https://warthogs.atlassian.net/browse/WD-13123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://ubuntu-com-14149.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/9131)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

petesfrench commented:
This is awesome. Should definitely dial the Vanilla folk in on this if you haven't already. If we align our practices early on it could save some time down the line.

Bit of a nit pick, but the quote in the design is just using regular quotes:
![image](https://github.com/user-attachments/assets/4c7089f9-ad4f-4cfb-95d2-e16f9ded5d10)

petesfrench commented on templates/case-studies/templates/_macros-image.jinja:
```diff
@@ -0,0 +1,18 @@
+    <div class="u-fixed-width">
+      <div class="row">
+        <div class="u-align--center u-vertically-center u-no-padding--bottom p-image-container--cinematic">
+          <img src={{ image_url }} alt={{ caption }} />
```
I think we need to include another param for the alt text, as in the example at `/case-studies` the alt is the copyright information.

petesfrench commented on templates/case-studies/templates/_macros-image.jinja:
```diff
@@ -0,0 +1,18 @@
+
+{%- macro image(image_url, caption) -%}
+  <section class="p-strip">
+    <div class="u-fixed-width">
+      <div class="row">
+        <div class="u-align--center u-vertically-center u-no-padding--bottom p-image-container--cinematic">
```
```suggestion
      <div class="row">
        <div class="col-12 u-align--center u-vertically-center u-no-padding--bottom p-image-container--cinematic">
```

petesfrench replied:
`u-fixed-width` accomplishes the same as `row`, so can be omitted here. But make sure you wrap children of a `row` in a `col`.

Another option would be to wrap the image in its own `u-fixed-width` and the caption in a `row > col`

britneywwc replied:
Thanks for the catch, I'll remove `u-fixed-width` since it is redundant in this case

petesfrench commented on templates/case-studies/templates/_macros-quote.jinja:
```diff
@@ -0,0 +1,28 @@
+            </blockquote>
+          </div>
+          <div class="col-3">
+            <p>
+              <br class="u-hide--large" />
```
```suggestion
            <br class="u-hide--large" />
            <p>
              
```

petesfrench commented on templates/case-studies/templates/_macros-quote.jinja:
```diff
@@ -0,0 +1,28 @@
+        <hr />
+        <div class="row">
+          <div class="col-6">
+            <blockquote class="p-pull-quote" style="margin: 0;">
```
```suggestion
            <blockquote class="p-pull-quote" style="margin: 0;">
```
is this `style="margin: 0;"` something that could be put in a class?

britneywwc replied:
I created a class for p-pull-quote for this style and also for the quotes colour

petesfrench commented on templates/case-studies/index.html:
```diff
@@ -6,41 +6,54 @@
-                      )
+  {{ text_template.text(
+      title="You can’t explore space without infrastructure.",
+      description="It’s not just expert personnel, satellites and spacecraft: it takes mountains of software, computer systems, data and networks to get these missions off the ground. These missions are often unique – testing the limits of our technological know-how, having never been done before with each mission presenting its own unique challenges. Building and deploying these systems on a mis ...
```
I wonder if there is a better way to denote different paragraphs. The double break `<br/><br/>` feels a little strange to me and might be too much to ask from content writers

britneywwc replied:
I implemented it as a list for now so content writers can input each paragraph as a list item. I tried doing a new paragraph but the spacing doesn't seem right so stuck with the double break in the macros for now. Let me know what you think!

petesfrench reviewed (commented):
Looks great! I've left a couple comments and a couple questions. I am lacking context on this work so feel free to ignore anything that has already been resolved/discussed :^)

britneywwc commented:
Thanks @petesfrench, I've addressed all of your comments and suggestion :) Could you have another look please?
Re Vanilla, I am planning on getting their input once all the macro PRs are merged.

petesfrench reviewed (approved):
Nice! LGTM :ship:
